ibis/backends/sql/compilers/mssql.py

```diff
@@ -404,6 +400,8 @@ def visit_Cast(self, op, *, arg, to):
             return arg
         elif from_.is_integer() and to.is_timestamp():
             return self.f.dateadd(self.v.s, arg, "1970-01-01 00:00:00")
+        elif from_.is_integer() and to.is_interval():
+            return sge.Interval(this=arg, unit=self.v[to.unit.singular])
```

Comment:
Since this is such a common implementation I'm going to try and encode the logic in the base implementation of `visit_Cast`